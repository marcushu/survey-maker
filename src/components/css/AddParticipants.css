#instructions {
    color: rgb(56, 56, 56);
    padding: 15px;
    font-size: 12px;
    font-style: italic;
}

#surveyTitle {
    color: rgb(1, 85, 134);
    font-size: 20px;
}

#backButton {
    color:rgb(1, 85, 134);    
    background-color:  white;
    border-style: solid;
    border-color: rgb(1, 85, 134);   
    font-size: 18px;
    font-weight: bold;
}

#addButton {
    background-color: rgb(1, 85, 134);
    border: none;
    font-size: 20px;
    color: white;
    padding: 8px;
    margin: 12px;
}

textarea {
    width: 300px;
}

