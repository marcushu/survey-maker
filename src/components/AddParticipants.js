import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import QuestionaireserviceMGO from '../js/QuestionaireServiceMGO';
import './css/AddParticipants.css';
import PropTypes from 'prop-types';


export default function AddParticipants({ surveyName, hideMe }) {
  const [participants, setparticipants] = useState("");


  const addParticipants = () => {
    if (!participants)
      alert("Please add a name/names.");
    else {
      const cleanNames = participants
        .split('\n')
        .join()
        .split(',')
        .filter(name => name.trim());

      QuestionaireserviceMGO.addMultSurveyTakers(surveyName, cleanNames)
        .then(() => {
          setparticipants("");
          alert("New Participants Added");
        })
        .catch(err => console.error(err));
    }
  }


  return (
    <>
      <Row>
        <Col className="d-flex justify-content-center">
          <div id="surveyTitle">{surveyName}</div>
        </Col>
      </Row>
      <Row >
        <Col className="d-flex justify-content-center">
          <div id="instructions">
            Add a <span style={{ fontWeight: "bold" }}>comma separated</span> list of new survey participants
            </div>
        </Col>
      </Row>
      <Row>
        <Col className="d-flex justify-content-center">
          <textarea rows={10}
            value={participants}
            onChange={e => setparticipants(e.target.value)} />
        </Col>
      </Row>
      <Row>
        <Col className="text-center">
          <button id="addButton" onClick={hideMe}>&lt;&lt;</button>
          <button id="addButton" onClick={addParticipants}>Add Participants</button>
        </Col>
      </Row>
    </>
  )
}

AddParticipants.propTypes = {
  surveyname: PropTypes.string,
  hideMe: PropTypes.func
}