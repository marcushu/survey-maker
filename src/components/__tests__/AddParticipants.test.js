import React from 'react';
import { render } from '@testing-library/react';
import AddParticipants from '../AddParticipants';

describe("AddParcipants", () => {
    const mockFunc = jest.fn();

    it("renders", () => {
        const { asFragment } = render(
            <AddParticipants
                surveyName="Mock Survey"
                hideMe={mockFunc}/> );

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with name", () => {
        const { getByText } = render(
            <AddParticipants
                surveyName="Mock Survey"
                hideMe={mockFunc}/> );

        expect(getByText(/Mock Survey/i)).toBeInTheDocument();
    })
})