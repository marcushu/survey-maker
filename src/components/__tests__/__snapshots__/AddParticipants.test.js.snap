// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddParcipants renders 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="d-flex justify-content-center col"
    >
      <div
        id="surveyTitle"
      >
        Mock Survey
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="d-flex justify-content-center col"
    >
      <div
        id="instructions"
      >
        Add a 
        <span
          style="font-weight: bold;"
        >
          comma separated
        </span>
         list of new survey participants
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="d-flex justify-content-center col"
    >
      <textarea
        rows="10"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="text-center col"
    >
      <button
        id="addButton"
      >
        &lt;&lt;
      </button>
      <button
        id="addButton"
      >
        Add Participants
      </button>
    </div>
  </div>
</DocumentFragment>
`;
